flamingo/cat_interface.py

```diff
@@ -245,24 +225,38 @@ def compute_bulkiness(smiles: pd.Series, opts: Options) -> np.ndarray:
     return results
 
 
-def compute_cosmo_rs(smiles: pd.Series, opts: Options) -> pd.DataFrame:
-    """Compute bulkiness using CAT.
-
-    It creates several instances of CAT using multiprocessing.
+def compute_cosmo_rs(molecules: pd.DataFrame, solvents: Dict[str, str], workdir: str) -> pd.DataFrame:
+    """Compute Cosmo Rs properties using CAT.
 
     Parameters
     ----------
-    smiles
-        Pandas.Series with the smiles to compute
-    opts
-        Options to call CAT
+    molecules
+        Pandas.DataFrame with the smiles to compute
+    solvents
+        Dictionary with the Paths to the solvents data
+    workdir
+        Directory to write the temporal results
 
     Returns
     -------
     pandas.DataFrame 
         Values
 
+    example:
+    >>> smiles = pd.Series(['CO'])
+    >>> solvents = {"hexane": "$AMSRESOURCES/ADFCRS/Hexane.coskf",
+                    "toluene": "$AMSRESOURCES/ADFCRS/Toluene.coskf"}
+    >>> compute_cosmo_rs(smiles, solvents, Path("."))
+
     """
-    results = map_reduce(smiles, opts, compute_batch_cosmo_rs, pd.concat)
+    output_dir = Path(tempfile.mkdtemp(prefix="cosmo_rs_", dir=workdir))
+    try:
+        rs = run_fast_sigma(molecules.smiles, solvents, output_dir=output_dir, return_df=True)
+        molecules = pd.merge(molecules, rs, left_on="smiles", right_index=True)
+    except RuntimeError:
+        pass
+    finally:
+        shutil.rmtree(output_dir)
```

Comment:
```suggestion
    with tempfile.TemporaryDirectory(prefix="cosmo_rs_", dir=workdir) as output_dir:
        rs = run_fast_sigma(molecules.smiles, solvents, output_dir=output_dir, return_df=True)
        molecules = pd.merge(molecules, rs, left_on="smiles", right_index=True)
```
This would be a bit more compact, assuming that neither `run_fast_sigma` nor `pd.merge` can raise a `RuntimeError` (this is assumption true for the former; not sure about `pd.merge`).

Reply:
Nice!